modules/sfp_zetalytics.py

```diff
@@ -60,14 +62,21 @@ def watchedEvents(self):
         return ["INTERNET_NAME", "DOMAIN_NAME", "EMAILADDR"]
 
     def producedEvents(self):
-        return ["INTERNET_NAME", "AFFILIATE_DOMAIN_NAME"]
+        return ["INTERNET_NAME", "AFFILIATE_DOMAIN_NAME", "INTERNET_NAME_UNRESOLVED"]
 
     def emit(self, etype, data, pevent, notify=True):
         evt = SpiderFootEvent(etype, data, self.__name__, pevent)
         if notify:
             self.notifyListeners(evt)
         return evt
 
+    def verify_emit_internet_name(self, hostname, pevent):
+        if self.opts["verify"] and not self.sf.resolveHost(hostname):
+            self.sf.debug(f"Host {hostname} could not be resolved")
+            self.emit("INTERNET_NAME_UNRESOLVED", hostname, pevent)
+        else:
+            self.emit("INTERNET_NAME", hostname, pevent)
+
```

Comment:
Rather than duplicate the `emit` function with a `verify_emit_internet_name` variant, I think from a design perspective it makes more sense to have the resolution logic outside of the `emit` function.

Reply:
```
        if self.opts["verify"] and not self.sf.resolveHost(hostname):
            self.sf.debug(f"Host {hostname} could not be resolved")
            self.emit("INTERNET_NAME_UNRESOLVED", hostname, pevent)
        else:
            self.emit("INTERNET_NAME", hostname, pevent)
```
This logic is used twice in functions below. Do you suggest to have 2 copies of it instead of shorter `verify_emit_internet_name` call?

Reply:
> This logic is used twice in functions below. Do you suggest to have 2 copies of it instead of shorter `verify_emit_internet_name` call?

Yes, unless you can think of a good reason to include it. I'm ok with a small amount of code duplication if it makes the module more intuitive.

However, this module is [missing some code patterns for determining whether the identified data is within the target scope](https://github.com/smicallef/spiderfoot/pull/992#issuecomment-690991443) using `self.getTarget()`. Adding in these code patterns may increase the code duplication sufficiently such that this helper function is warranted.